src/OrchardCore/OrchardCore.Abstractions/Modules/Builder/OrchardCoreBuilder.cs

```diff
@@ -14,9 +15,14 @@ public class OrchardCoreBuilder
 
         public OrchardCoreBuilder(IServiceCollection services)
         {
+            Configuration = (IConfiguration)services
+                .FirstOrDefault(d => d.ServiceType == typeof(IConfiguration))
+                .ImplementationFactory(null);
```

Comment:
Does FirstOrDefault return Null here?

Reply:
It should not, both the generic host and web host register a singleton with an `ImplementationFactory` that just returns the `IConfiguration` instance that has been built by the `HostBuilder` or the `WebHostBuilder`, and without using the passed service provider (so here we can pass null).

So here it may fail on an NRE, but only if there is an AspnetCore breaking change